crypto/provider_conf.c

```diff
@@ -364,22 +364,24 @@ static int provider_conf_load(OSSL_LIB_CTX *libctx, const char *name,
         OSSL_PROVIDER_INFO entry;
 
         memset(&entry, 0, sizeof(entry));
-        ok = 1;
+        ok = 0;
         if (name != NULL) {
             entry.name = OPENSSL_strdup(name);
-            if (entry.name == NULL)
-                ok = 0;
+            if (entry.name != NULL)
+                ok = 1;
         }
-        if (ok && path != NULL) {
-            entry.path = OPENSSL_strdup(path);
-            if (entry.path == NULL)
-                ok = 0;
+        if (ok) {
+            ok = 0;
```

Comment:
This looks wrong, as I think it is valid for a path to be NULL

Reply:
Thanks. I have submitted a new commit and remove the above changes.